app/question.js

```diff
@@ -60,23 +65,38 @@ define(['jquery', 'app/messages', 'app/config', 'app/string'], function ($, mess
   // Handle how to move to the next question
   // depending on the tutorial status.
   function nextQuestion() {
-    if (!inTutorial) {
-      items.splice(currentItemIndex,1);
-      currentItemIndex %= items.length;
-    } else {
+    if (inTutorial) {
       currentItemIndex++;
       inTutorial = checkTutorialStatus();
+    } else if (answerWasCorrect) {
+      items.splice(currentItemIndex, 1);
+      currentItemIndex %= items.length;
+    } else {
+      currentItemIndex = (currentItemIndex + 1) % items.length;
     }
   }
 
   function showTutorialInstruction() {
     $("#question").append("<br><b>Type the answer:</b> " + items[currentItemIndex].item_answer);
```

Comment:
To me it makes more sense to have this as a span in the view that you show and hide? That way you keep your view separate from the controller.